### ISSUE TITLE
Spatial attribute not converted error

### ISSUE THREAD

**Issue by cccaballero** [sjaakp/yii2-spatial#5]
Hi, Thanks for such a useful extension!!

I am using this exact query:

```
$crowds = Crowd::find()
            ->joinWith('reports')
            ->select(['crowd.*', 'report.created_at'])
            ->andWhere(['>=', 'reports_count', Crowd::$REPORTS_TRUST_NUMBER])
            ->andWhere(['>=', 'report.created_at', new Expression('DATE_SUB(NOW(), INTERVAL 12 HOUR)')])->all();
```

and I am getting an "Spatial attribute not converted" error, `Crowd` is an instance of `sjaakp\spatial\ActiveRecord` and my `find()` function is returning an instance of `sjaakp\spatial\ActiveQuery`

For making it simpler, we can reduce the problematic query to:

```
$crowds = Crowd::find()
            ->joinWith('reports')
            ->select(['crowd.*'])->all();
```

I need to use `select()` because both `crowd` and `report` tables have a `location` field and if `select()` is not used I get a "Integrity constraint violation" because `location` field is ambiguous.

**Comment by sjaakp** [comment]
Hmm, beats me. Is Report a spatial\ActiveRecord as well? I assume its location field is spatial.

 

  _____  

Sjaak Priester

sjaak@sjaakpriester.nl <mailto:sjaak@sjaakpriester.nl> 

 

Van: Carlos Cesar Caballero Díaz <notifications@github.com> 
Verzonden: maandag 30 maart 2020 07:43
Aan: sjaakp/yii2-spatial <yii2-spatial@noreply.github.com>
CC: Subscribed <subscribed@noreply.github.com>
Onderwerp: [sjaakp/yii2-spatial] Spatial attribute not converted error (#5)

 

Hi, Thanks for such a useful extension!!

I am using this exact query:

$crowds = Crowd::find()
            ->joinWith('reports')
            ->select(['crowd.*', 'report.created_at'])
            ->andWhere(['>=', 'reports_count', Crowd::$REPORTS_TRUST_NUMBER])
            ->andWhere(['>=', 'report.created_at', new Expression('DATE_SUB(NOW(), INTERVAL 12 HOUR)')])->all();

and I am getting an "Spatial attribute not converted" error, Crowd is an instance of sjaakp\spatial\ActiveRecord and my find() function is returning an instance of sjaakp\spatial\ActiveQuery

For making it simpler, we can reduce the problematic query to:

$crowds = Crowd::find()
            ->joinWith('reports')
            ->select(['crowd.*'])->all();

I need to use select() because both crowd and report tables have a location field and if select() is not used I get a "Integrity constraint violation" because location field is ambiguous.

**Comment by cccaballero** [comment]
yes, both are `spatial\ActiveRecord ` instances. Right now I solved my issue using two querys, one for obtaining the `report.crowd_id` fields, and the other one for obtaining the `Crowds` who match.
